domains/client/domain-operator/src/domain_worker.rs

```diff
@@ -123,15 +124,16 @@ pub(super) async fn start_worker<
 
     if let Some(operator_id) = maybe_operator_id {
         info!("👷 Running as Operator[{operator_id}]...");
-        let mut new_slot_notification_stream = pin!(new_slot_notification_stream);
+        let mut latest_slot_notification_stream =
+            LatestItemStream::new(new_slot_notification_stream);
         let mut acknowledgement_sender_stream = pin!(acknowledgement_sender_stream);
         loop {
             tokio::select! {
                 // Ensure any new slot/block import must handle first before the `acknowledgement_sender_stream`
                 // NOTE: this is only necessary for the test.
                 biased;
 
-                Some((slot, proof_of_time)) = new_slot_notification_stream.next() => {
+                Some((slot, proof_of_time)) = latest_slot_notification_stream.next() => {
```

Comment:
The adapter below handles the stream terminating, but this `select!` ignores termination (it doesn’t check for `None`, so the next branch will be executed).

Is that the intended behaviour, or should the loop break instead?

Reply:
We always expect the pot slot worker to be sending the info. If the slot worker stream is closed, then essential task is failed anyway since the entire process stops. Its fine to leave it as is